### PR TITLE
Encrypted attributes

### PR DESCRIPTION
**Pull request by mochetts**
German laws determine that sensitive data must not be stored in plain text, but encrypted. The main purpose of it is that in case the database access is compromised, then the data would be unreadable.

As an example, credit card numbers, emails, first and last names, social security numbers, etc.

For this, we extended the Eloquent/Model.php class in order to address encryptable attributes. So whenever an attribute is "set" we encrypt it and whenever is "get" we decrypt it.

Would be nice to have this in the base-line of Laravel.

In case there's a better alternative on how to implement this, I'm happy to code it.

Thanks.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -2390,6 +2397,15 @@ public function attributesToArray()
 	{
 		$attributes = $this->getArrayableAttributes();
 
+		// If an attribute is listed as to "encrypt", we'll decrypt it using Crypt facade.
+		foreach ($attributes as $key => $value)
+        {
```
Comment (jtraub) [inline src/Illuminate/Database/Eloquent/Model.php]: Obviously you've misindented the code here

Comment (mmochetti) [inline src/Illuminate/Database/Eloquent/Model.php]: Let me check, I'm using sublime + smartgit and the identation was fine in both tools.

**Comment by Anahkiasen** [comment]
Think this was rejected already [just a few days ago](https://github.com/laravel/framework/pull/8704)

**Comment by mmochetti** [comment]
I read the thread. Two things to point out:

1) The code is slightly different. But both persue the same objective.
2) Seems it's becoming a recurrent issue/requirement.

I don't mind having it on my own Laravel fork, but seems to be something that many web apps require around there. Just googling it gives you an idea.

I suggested the pull request because as I said, it's becoming a recurrent issue and I think that's why date mutation, guarded attributes, native type casting and fillable attributes were also added at the time.

Cheers.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -2390,6 +2397,15 @@ public function attributesToArray()
 	{
 		$attributes = $this->getArrayableAttributes();
 
+		// If an attribute is listed as to "encrypt", we'll decrypt it using Crypt facade.
+		foreach ($attributes as $key => $value)
+		{
+			if (in_array($key, $this->encrypt) && $value)
+			{
+				$attributes[$key] = \Crypt::decrypt($value);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: You can't use facades like thus,

**Comment by GrahamCampbell** [comment]
:-1:

File: src/Illuminate/Database/Eloquent/Model.php (hunk 2 above)
Comment (mmochetti) [inline src/Illuminate/Database/Eloquent/Model.php]: Good to learn new things every day. Can I ask why? Thanks .

Comment (franzliedke) [inline src/Illuminate/Database/Eloquent/Model.php]: They're just simple accessors for instances stored in the IoC container, and are only supposed to make application code easier. They're not used anywhere in Laravel's core.

**Pull request closed without merging**